// @frontend/wknd-components/dist/src/Tooltip/Tooltip.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooltipText = exports.StyledTooltip = exports.TooltipContainer = exports.TooltipWrapper = exports.animation = exports.tooltipPositions = void 0;
var styled_components_1 = __importStar(require("styled-components"));
var styleguide_1 = require("../styleguide");
exports.tooltipPositions = {
    "top-start": "\n\t\tbottom: 100%;\n\t\tmargin-bottom: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    top: "\n\t\tbottom: 100%;\n\t\tright: 50%;\n\t\ttransform: translate(50%, 0);\n\t\tmargin-bottom: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "top-end": "\n\t\tbottom: 100%;\n\t\tright: 0;\n\t\tmargin-bottom: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "bottom-start": "\n\t\tleft: 0;\n\t\tmargin-top: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    bottom: "\n\t\tright: 50%;\n\t\ttransform: translate(50%, 0);\n\t\tmargin-top: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "bottom-end": "\n\t\tright: 0;\n\t\tmargin-top: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "left-start": "\n\t\tright: 100%;\n\t\ttop: 0;\n\t\tmargin-right: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    left: "\n\t\tright: 100%;\n\t\ttransform: translate(0, 50%);\n\t\tbottom: 50%;\n\t\tmargin-right: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "left-end": "\n\t\tright: 100%;\n\t\tbottom: 0;\n\t\tmargin-right: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "right-start": "\n\t\tleft: 100%;\n\t\ttop: 0;\n\t\tmargin-left: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    right: "\n\t\tleft: 100%;\n\t\ttransform: translate(0, 50%);\n\t\tbottom: 50%;\n\t\tmargin-left: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
    "right-end": "\n\t\tleft: 100%;\n\t\tbottom: 0;\n\t\tmargin-left: ".concat(styleguide_1.spacingMap.sm, ";\n\t"),
};
exports.animation = (0, styled_components_1.keyframes)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tfrom {\n\t\topacity: 0.6;\n\t}\n\n\tto {\n\t\topacity: 1;\n\t}\n"], ["\n\tfrom {\n\t\topacity: 0.6;\n\t}\n\n\tto {\n\t\topacity: 1;\n\t}\n"])));
exports.TooltipWrapper = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\tposition: relative;\n\tdisplay: inline-block;\n"], ["\n\tposition: relative;\n\tdisplay: inline-block;\n"])));
exports.TooltipContainer = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tposition: relative;\n\tdisplay: inline-flex;\n\tmax-width: max-content;\n\tcursor: pointer;\n"], ["\n\tposition: relative;\n\tdisplay: inline-flex;\n\tmax-width: max-content;\n\tcursor: pointer;\n"])));
exports.StyledTooltip = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tposition: absolute;\n\tz-index: ", ";\n\tcolor: ", ";\n\ttext-align: left;\n\tbackground-color: ", ";\n\tborder-radius: 4px;\n\tbox-shadow: 0 2px 2px rgba(0, 0, 0, 0.08),\n\t0 4px 4px rgba(0, 0, 0, 0.08),\n\t0 8px 8px rgba(0, 0, 0, 0.08),\n\t0 16px 32px rgba(0, 0, 0, 0.08);\n\t", ";\n\tanimation: ", " 0.2s linear;\n\tpadding: ", " ", ";\n\tbox-sizing: border-box;\n\t", ";\n"], ["\n\tposition: absolute;\n\tz-index: ", ";\n\tcolor: ", ";\n\ttext-align: left;\n\tbackground-color: ", ";\n\tborder-radius: 4px;\n\tbox-shadow: 0 2px 2px rgba(0, 0, 0, 0.08),\n\t0 4px 4px rgba(0, 0, 0, 0.08),\n\t0 8px 8px rgba(0, 0, 0, 0.08),\n\t0 16px 32px rgba(0, 0, 0, 0.08);\n\t", ";\n\tanimation: ", " 0.2s linear;\n\tpadding: ", " ", ";\n\tbox-sizing: border-box;\n\t", ";\n"])), function (_a) {
    var zIndex = _a.zIndex;
    return zIndex;
}, styleguide_1.colors.white, styleguide_1.colors.black, function (_a) {
    var isLong = _a.isLong;
    return isLong && "min-width: ".concat((styleguide_1.widthSizes.sr));
}, exports.animation, styleguide_1.spacingMap.xxs, styleguide_1.spacingMap.xs, function (_a) {
    var position = _a.position;
    return position && exports.tooltipPositions[position];
});
exports.TooltipText = styled_components_1.default.p(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\twidth: ", ";\n\tfont-family:  ", ";\n\tfont-size: ", ";\n\tline-height: ", ";\n\tmargin: 0;\n\t", ";\n"], ["\n\twidth: ", ";\n\tfont-family:  ", ";\n\tfont-size: ", ";\n\tline-height: ", ";\n\tmargin: 0;\n\t", ";\n"])), styleguide_1.widthSizes.full, styleguide_1.displayFontSizes.caption.$fontFamily, styleguide_1.displayFontSizes.caption.$fontSize, styleguide_1.displayFontSizes.caption.lineHeight, function (_a) {
    var isLong = _a.isLong;
    return !isLong && "white-space: nowrap";
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
