// @frontend/wknd-components/dist/src/Toast/ToastNotification.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var IconButton_1 = __importDefault(require("../Button/IconButton"));
var Icon_1 = require("../Icon/Icon");
var styleguide_1 = require("../styleguide");
var Toast_styles_1 = require("./Toast.styles");
var ToastContext_1 = require("./ToastContext");
var toastVariantOptions = {
    Default: {
        iconName: "Info",
        color: styleguide_1.colors.white,
    },
    Warning: {
        iconName: "AlertTriangle",
        color: styleguide_1.colors.black,
    },
    Error: {
        iconName: "AlertOctagon",
        color: styleguide_1.colors.black,
    },
    Validation: {
        iconName: "Check",
        color: styleguide_1.colors.black,
    },
};
var ToastNotification = function (_a) {
    var id = _a.id, dataQA = _a.dataQA, className = _a.className, text = _a.text, _b = _a.variant, variant = _b === void 0 ? "Default" : _b, _c = _a.showIcon, showIcon = _c === void 0 ? true : _c, trailingLinkText = _a.trailingLinkText, trailingLinkAction = _a.trailingLinkAction, iconName = _a.iconName, iconColor = _a.iconColor, textColor = _a.textColor, backgroundColor = _a.backgroundColor, _d = _a.hasTimeout, hasTimeout = _d === void 0 ? true : _d, _e = _a.timeout, timeout = _e === void 0 ? 3000 : _e, onTimeout = _a.onTimeout, onDismiss = _a.onDismiss;
    var removeNotification = (0, ToastContext_1.useToast)().removeNotification;
    var trailingButtonRef = (0, react_1.useRef)(document.createElement("button"));
    var _f = (0, react_1.useState)(0), trailingButtonWidth = _f[0], setTrailingButtonWidth = _f[1];
    (0, react_1.useLayoutEffect)(function () {
        setTrailingButtonWidth(trailingButtonRef.current.offsetWidth);
    }, [trailingButtonWidth]);
    (0, react_1.useEffect)(function () {
        if (hasTimeout) {
            var timeoutId_1 = setTimeout(function () {
                removeNotification(id);
                if (onTimeout) {
                    onTimeout();
                }
            }, timeout);
            return function () { return clearTimeout(timeoutId_1); };
        }
        return;
    }, [removeNotification, hasTimeout, timeout, onTimeout, id]);
    var onDismissNotification = function () {
        removeNotification(id);
        if (onDismiss) {
            onDismiss();
        }
    };
    return (react_1.default.createElement(react_1.default.Fragment, null, trailingButtonWidth > -1 && (react_1.default.createElement(Toast_styles_1.StyledToast, { key: id, "data-qa": dataQA, className: className, variant: variant, backgroundColor: backgroundColor, role: "status" },
        showIcon && (react_1.default.createElement(Toast_styles_1.IconWrapper, null,
            react_1.default.createElement(Icon_1.Icon, { "data-qa": "".concat(dataQA, "-icon"), role: "presentation", name: iconName ||
                    toastVariantOptions[variant].iconName, size: 18, strokeWidth: "2", color: iconColor ||
                    toastVariantOptions[variant].color }))),
        react_1.default.createElement(Toast_styles_1.StyledToastTypography, { dataQA: "".concat(dataQA, "-typography"), variant: "bodyCopy", component: "span", color: textColor || toastVariantOptions[variant].color, title: text }, text),
        trailingLinkText && (react_1.default.createElement(Toast_styles_1.TrailingLinkButton, { ref: trailingButtonRef, dataQA: "".concat(dataQA, "-trailing-link"), onClick: trailingLinkAction, buttonText: trailingLinkText, variant: variant })),
        react_1.default.createElement(Toast_styles_1.DismissButtonWrapper, { variant: variant },
            react_1.default.createElement(IconButton_1.default, { dataQA: "".concat(dataQA, "-dismiss-button"), "aria-label": "dismiss", icon: "X", onClick: onDismissNotification, variant: "tertiary" }))))));
};
exports.default = ToastNotification;
