// @frontend/wknd-components/dist/src/Toast/ToastContext.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteNotification = exports.triggerNotification = exports.useToast = exports.ToastProvider = void 0;
var react_1 = require("react");
var react_2 = __importDefault(require("react"));
var uuid_1 = __importDefault(require("../utils/uuid"));
var ToastContext = (0, react_1.createContext)(undefined);
var appendNotificationExport;
var removeNotificationExport;
var toastReducer = function (state, action) {
    switch (action.type) {
        case "APPEND_NOTIFICATION":
            return __spreadArray(__spreadArray([], state, true), [action.payload], false);
        case "REMOVE_NOTIFICATION":
            return state.filter(function (notification) { return notification.id !== action.payload; });
        default:
            return state;
    }
};
var ToastProvider = function (_a) {
    var children = _a.children;
    var _b = (0, react_1.useReducer)(toastReducer, []), notifications = _b[0], dispatch = _b[1];
    var appendNotification = (0, react_1.useCallback)(function (notification) {
        var id = (0, uuid_1.default)();
        dispatch({
            type: "APPEND_NOTIFICATION",
            payload: __assign(__assign({}, notification), { id: id }),
        });
    }, []);
    var removeNotification = (0, react_1.useCallback)(function (id) {
        dispatch({ type: "REMOVE_NOTIFICATION", payload: id });
    }, []);
    appendNotificationExport = appendNotification;
    removeNotificationExport = removeNotification;
    return (react_2.default.createElement(ToastContext.Provider, { value: { notifications: notifications, appendNotification: appendNotification, removeNotification: removeNotification } }, children));
};
exports.ToastProvider = ToastProvider;
// useToast is a custom hook to access the toast context and should be used within a React component.
var useToast = function () {
    var context = (0, react_1.useContext)(ToastContext);
    if (!context) {
        throw new Error("useToast must be used within a ToastProvider");
    }
    return context;
};
exports.useToast = useToast;
// triggerNotification is a function to trigger a toast notification outside of React components eg: errorHandler.ts.
var triggerNotification = function (notification) {
    if (!appendNotificationExport) {
        throw new Error("triggerNotification must be used within a ToastProvider");
    }
    appendNotificationExport(notification);
};
exports.triggerNotification = triggerNotification;
// deleteNotification is a function to remove a toast notification outside of React components.
var deleteNotification = function (id) {
    if (!removeNotificationExport) {
        throw new Error("deleteNotification must be used within a ToastProvider");
    }
    removeNotificationExport(id);
};
exports.deleteNotification = deleteNotification;
