// @frontend/wknd-components/dist/src/DatePicker/DatePicker.defaults.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_SHORTCUTS = exports.defaultShortcutsList = exports.today = exports.getDefaultDates = void 0;
var addYears_1 = __importDefault(require("date-fns/addYears"));
var endOfDay_1 = __importDefault(require("date-fns/endOfDay"));
var lastDayOfMonth_1 = __importDefault(require("date-fns/lastDayOfMonth"));
var lastDayOfQuarter_1 = __importDefault(require("date-fns/lastDayOfQuarter"));
var startOfDay_1 = __importDefault(require("date-fns/startOfDay"));
var startOfMonth_1 = __importDefault(require("date-fns/startOfMonth"));
var startOfQuarter_1 = __importDefault(require("date-fns/startOfQuarter"));
var startOfYear_1 = __importDefault(require("date-fns/startOfYear"));
var subDays_1 = __importDefault(require("date-fns/subDays"));
var subMonths_1 = __importDefault(require("date-fns/subMonths"));
var subQuarters_1 = __importDefault(require("date-fns/subQuarters"));
var subYears_1 = __importDefault(require("date-fns/subYears"));
var getDefaultDates = function () { return ({
    defaultMinDate: (0, subYears_1.default)(new Date().setHours(0, 0, 0, 0), 5),
    defaultMaxDate: (0, addYears_1.default)(new Date().setHours(0, 0, 0, 0), 5),
}); };
exports.getDefaultDates = getDefaultDates;
exports.today = (0, startOfDay_1.default)(new Date());
/**
 * NOTE: If a change is made here, make sure it gets replicated in `__mocks__/DatePicker.defaults`
 */
exports.defaultShortcutsList = {
    today: {
        label: "Today",
        getDates: function () { return [exports.today, (0, endOfDay_1.default)(exports.today)]; },
        id: "today",
    },
    yesterday: {
        label: "Yesterday",
        getDates: function () { return [(0, subDays_1.default)(exports.today, 1), (0, endOfDay_1.default)((0, subDays_1.default)(exports.today, 1))]; },
        id: "yesterday",
    },
    lastSevenDays: {
        label: "Last 7 days",
        getDates: function () { return [(0, subDays_1.default)(exports.today, 7), (0, subDays_1.default)(exports.today, 1)]; },
        id: "last-seven-days",
    },
    lastThirtyDays: {
        label: "Last 30 days",
        getDates: function () { return [(0, subDays_1.default)(exports.today, 30), (0, subDays_1.default)(exports.today, 1)]; },
        id: "last-thirty-days",
    },
    lastNinetyDays: {
        label: "Last 90 days",
        getDates: function () { return [(0, subDays_1.default)(exports.today, 90), (0, subDays_1.default)(exports.today, 1)]; },
        id: "last-ninety-days",
    },
    lastMonth: {
        label: "Last month",
        getDates: function () {
            var lastMonthStartDate = (0, startOfMonth_1.default)((0, subMonths_1.default)(exports.today, 1));
            var lastMonthEndDate = (0, lastDayOfMonth_1.default)((0, subMonths_1.default)(exports.today, 1));
            return [lastMonthStartDate, lastMonthEndDate];
        },
        id: "last-month",
    },
    lastQuarter: {
        label: "Last quarter",
        getDates: function () {
            var lastQuarterStartDate = (0, startOfQuarter_1.default)((0, subQuarters_1.default)(exports.today, 1));
            var lastQuarterEndDate = (0, lastDayOfQuarter_1.default)((0, subQuarters_1.default)(exports.today, 1));
            return [lastQuarterStartDate, lastQuarterEndDate];
        },
        id: "last-quarter",
    },
    yearToDate: {
        label: "Year to date",
        getDates: function () { return [(0, startOfYear_1.default)(exports.today), exports.today]; },
        id: "year-to-date",
    },
};
exports.DEFAULT_SHORTCUTS = [
    exports.defaultShortcutsList.today,
    exports.defaultShortcutsList.yesterday,
    exports.defaultShortcutsList.lastSevenDays,
    exports.defaultShortcutsList.lastThirtyDays,
    exports.defaultShortcutsList.lastNinetyDays,
    exports.defaultShortcutsList.lastMonth,
    exports.defaultShortcutsList.lastQuarter,
    exports.defaultShortcutsList.yearToDate,
];
