// @frontend/wknd-components/dist/src/DatePicker/DatePickerPopover.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var date_fns_1 = require("date-fns");
var Popover_1 = require("../Popover");
var DatePickerInput_1 = __importDefault(require("./DatePickerInput"));
var DatePickerPopoverContext_1 = require("./DatePickerPopoverContext");
var DateRangePickerInput_1 = __importDefault(require("./DateRangePickerInput"));
// offset used for the Input triggers
var INPUT_OFFSET = 16;
// offset used when the trigger is a button
var TRIGGER_OFFSET = 8;
// align offset when no inputLeftIcon
var INPUT_OFFSET_WITHOUT_LEFT_ICON = -12;
// align offset with inputLeftIcon
var INPUT_OFFSET_WITH_LEFT_ICON = -40;
var DatePickerPopover = function (_a) {
    var children = _a.children, trigger = _a.trigger, dataQA = _a.dataQA, _b = _a.closeOnDateSelected, closeOnDateSelected = _b === void 0 ? true : _b, datesSelected = _a.datesSelected, onDateChange = _a.onDateChange, isRange = _a.isRange, label = _a.label, dateFormat = _a.dateFormat, placeholder = _a.placeholder, minDate = _a.minDate, maxDate = _a.maxDate, inputLeftIcon = _a.inputLeftIcon, inputRightIcon = _a.inputRightIcon, $zIndex = _a.$zIndex;
    var _c = (0, react_1.useState)(false), isCalendarOpen = _c[0], setIsCalendarOpen = _c[1];
    // this is needed to prevent the trigger from opening the calendar again if it was clicked
    // the onPointerDownOutside gets called before the onClick
    var _d = (0, react_1.useState)(false), outsideClicked = _d[0], setOutSideClicked = _d[1];
    var datePickerInputRef = (0, react_1.useRef)(null);
    var endDateInputRef = (0, react_1.useRef)(null);
    var handleTriggerOnFocus = function () {
        setIsCalendarOpen(true);
    };
    var handleTriggerOnClick = function () {
        if (outsideClicked) {
            setOutSideClicked(false);
            return;
        }
        setIsCalendarOpen(true);
    };
    var closeCalendar = function () {
        setIsCalendarOpen(false);
    };
    var handleOnPointerDownOutside = function (e) {
        // if input was clicked we don't need to close and can allow it to stay open
        if (datePickerInputRef.current === e.target || endDateInputRef.current === e.target) {
            return;
        }
        setOutSideClicked(true);
        closeCalendar();
    };
    var preventAutoFocus = function (event) {
        if (!trigger && datePickerInputRef.current || endDateInputRef.current) {
            event.preventDefault();
        }
    };
    var handleOnOpenChange = function (open) { return !open && setOutSideClicked(false); };
    (0, react_1.useEffect)(function () {
        if (!closeOnDateSelected) {
            return;
        }
        if (isRange && datesSelected instanceof Array) {
            if (!(0, date_fns_1.isSameDay)(datesSelected[0], datesSelected[1]) && datesSelected[1]) {
                closeCalendar();
            }
            return;
        }
        if (datesSelected && datesSelected instanceof Date) {
            closeCalendar();
        }
    }, [datesSelected, closeOnDateSelected, isRange]);
    var datePickerPopoverContext = {
        closeCalendarPopover: closeCalendar,
    };
    // this is needed to account for the inputLeftIcon since the Popover trigger is the start Input which
    // is indented from the visual input border
    var getPopoverAlignOffset = function () {
        if (isRange && !trigger && !inputLeftIcon) {
            return INPUT_OFFSET_WITHOUT_LEFT_ICON;
        }
        // this accounts for the size of the icon plus the additional padding
        if (inputLeftIcon) {
            return INPUT_OFFSET_WITH_LEFT_ICON;
        }
        return undefined;
    };
    var renderInputTrigger = function () { return isRange ? (react_1.default.createElement(DateRangePickerInput_1.default, { dataQA: "range-input", selectedDates: datesSelected instanceof Array ? datesSelected : undefined, id: "range-input", label: label, ref: datePickerInputRef, endDateInputRef: endDateInputRef, onDateChange: onDateChange, placeholder: placeholder, minDate: minDate, maxDate: maxDate, inputLeftIcon: inputLeftIcon, inputRightIcon: inputRightIcon })) : (react_1.default.createElement(DatePickerInput_1.default, { selectedDate: datesSelected instanceof Date ? datesSelected : undefined, setIsCalendarOpen: setIsCalendarOpen, onDateChange: onDateChange, label: label, ref: datePickerInputRef, dataQA: dataQA, dateFormat: dateFormat, placeholder: placeholder, inputLeftIcon: inputLeftIcon, inputRightIcon: inputRightIcon })); };
    return (react_1.default.createElement(DatePickerPopoverContext_1.DatePickerPopoverContextProvider, { value: datePickerPopoverContext },
        react_1.default.createElement(Popover_1.Popover, { trigger: trigger || renderInputTrigger(), triggerOnClick: handleTriggerOnClick, triggerOnFocus: trigger ? undefined : handleTriggerOnFocus, ariaLabel: "datepicker popover", dataQA: dataQA, radixPopoverProps: {
                open: isCalendarOpen,
                onOpenChange: handleOnOpenChange,
            }, radixPopoverContentProps: {
                // this is needed so that the offset is set correctly when using the Input
                sideOffset: !trigger ? INPUT_OFFSET : TRIGGER_OFFSET,
                alignOffset: getPopoverAlignOffset(),
                onEscapeKeyDown: closeCalendar,
                onPointerDownOutside: handleOnPointerDownOutside,
                onOpenAutoFocus: preventAutoFocus,
                onCloseAutoFocus: preventAutoFocus,
            }, "$zIndex": $zIndex }, children)));
};
exports.default = DatePickerPopover;
