// @frontend/wknd-components/dist/src/Modal/ModalFooter/ModalFooter.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DarkSecondaryButton = exports.StyledButton = exports.FooterWrapper = exports.border = exports.defaultFooterDisplay = exports.largeFooterDisplay = void 0;
var styled_components_1 = __importStar(require("styled-components"));
var Button_1 = __importDefault(require("../../Button/Button"));
var styleguide_1 = require("../../styleguide");
exports.largeFooterDisplay = (0, styled_components_1.css)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tdisplay: flex;\n\tjustify-content: flex-end;\n\tgap: ", ";\n"], ["\n\tdisplay: flex;\n\tjustify-content: flex-end;\n\tgap: ", ";\n"])), styleguide_1.spacingMap.xs);
exports.defaultFooterDisplay = (0, styled_components_1.css)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\tdisplay: grid;\n\tgrid-auto-columns: auto;\n\tgrid-gap: ", ";\n"], ["\n\tdisplay: grid;\n\tgrid-auto-columns: auto;\n\tgrid-gap: ", ";\n"])), styleguide_1.spacingMap.xs);
exports.border = (0, styled_components_1.css)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tborder-top: 1px solid ", ";\n"], ["\n\tborder-top: 1px solid ", ";\n"])), styleguide_1.colors.lightGray);
exports.FooterWrapper = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tpadding: ", " ", " ", ";\n\t", ";\n\t// Divider for scrollable variant is just a top border on the FooterWrapper\n\t", ";\n"], ["\n\tpadding: ", " ", " ", ";\n\t", ";\n\t// Divider for scrollable variant is just a top border on the FooterWrapper\n\t", ";\n"])), styleguide_1.spacingMap.sm, styleguide_1.spacingMap.md, styleguide_1.spacingMap.md, function (_a) {
    var size = _a.size;
    return size === "large" ? exports.largeFooterDisplay : exports.defaultFooterDisplay;
}, function (_a) {
    var variant = _a.variant;
    return variant === "scrollable" && exports.border;
});
exports.StyledButton = (0, styled_components_1.default)(Button_1.default)(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\tgrid-row: 1;\n"], ["\n\tgrid-row: 1;\n"])));
exports.DarkSecondaryButton = (0, styled_components_1.default)(exports.StyledButton)(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\tbackground: ", ";\n\tcolor: ", ";\n\tborder: 1px solid ", ";\n"], ["\n\tbackground: ", ";\n\tcolor: ", ";\n\tborder: 1px solid ", ";\n"])), styleguide_1.colors.black, styleguide_1.colors.white, styleguide_1.colors.white);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6;
