// @frontend/wknd-components/dist/src/styleguide/displayFontSizes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var fontFamily_1 = __importDefault(require("./fontFamily"));
var fontWeight_1 = __importDefault(require("./fontWeight"));
/**
 * @description see https://www.figma.com/file/W7QEYEtmd4U0Y2SCsiKWIT/Global-Styles?node-id=2%3A4 for our Global
 * Typography styles
 */
var displayFontSizes = {
    displayXLarge: {
        $fontSize: "2.5rem",
        lineHeight: "2.75rem",
        $fontFamily: fontFamily_1.default.wunderbrick,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Large heroes or banners. Used sparingly",
    },
    displayLarge: {
        $fontSize: "2rem",
        lineHeight: "2.25rem",
        $fontFamily: fontFamily_1.default.wunderbrick,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Large metrics and banners",
    },
    displayMedium: {
        $fontSize: "1.625rem",
        lineHeight: "2rem",
        $fontFamily: fontFamily_1.default.wunderbrick,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Large metrics and banners",
    },
    displaySmall: {
        $fontSize: "1.25rem",
        lineHeight: "1.5rem",
        $fontFamily: fontFamily_1.default.wunderbrick,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Module and object headings",
    },
    metricsLarge: {
        $fontSize: "2rem",
        lineHeight: "2.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.light,
        usage: "Metrics and numbers",
    },
    metricsMedium: {
        $fontSize: "1.25rem",
        lineHeight: "1.5rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Metrics and numbers",
    },
    metricsSmall: {
        $fontSize: "0.875rem",
        lineHeight: "1.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Metrics and numbers in data tables",
    },
    headline: {
        $fontSize: "1.25rem",
        lineHeight: "1.5rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.bold,
        usage: "Page titles, used once per page",
    },
    headlineSmall: {
        $fontSize: "1rem",
        lineHeight: "1.5rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.bold,
        usage: "Subtitles, section titles",
    },
    subHeadline: {
        $fontSize: "0.75rem",
        lineHeight: "1rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.bold,
        usage: "Subsections, form group subsections",
    },
    bodyCopy: {
        $fontSize: "0.875rem",
        lineHeight: "1.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.normal,
        usage: "General body copy",
    },
    bodyCopyBold: {
        $fontSize: "0.875rem",
        lineHeight: "1.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.bold,
        usage: "Bold body copy",
    },
    bodyCopySmall: {
        $fontSize: "0.8125rem",
        lineHeight: "1.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Small body copy when necessary due to space constraints",
    },
    bodyCopySmallBold: {
        $fontSize: "0.8125rem",
        lineHeight: "1.25rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.bold,
        usage: "Bold small body copy",
    },
    caption: {
        $fontSize: "0.75rem",
        lineHeight: "1rem",
        $fontFamily: fontFamily_1.default.larsseit,
        $fontWeight: fontWeight_1.default.normal,
        usage: "Smaller subtext for captions and labels, usually paired with a subtle text color",
    },
};
exports.default = displayFontSizes;
