// @frontend/wknd-components/dist/src/styleguide/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.widthSizes = exports.tagMap = exports.spacingMap = exports.sizes = exports.fontWeight = exports.fontSizes = exports.fontFamily = exports.displayFontSizes = exports.colors = void 0;
var colors_1 = __importDefault(require("./colors"));
exports.colors = colors_1.default;
var displayFontSizes_1 = __importDefault(require("./displayFontSizes"));
exports.displayFontSizes = displayFontSizes_1.default;
var fontFamily_1 = __importDefault(require("./fontFamily"));
exports.fontFamily = fontFamily_1.default;
var fontSizes_1 = __importDefault(require("./fontSizes"));
exports.fontSizes = fontSizes_1.default;
var fontWeight_1 = __importDefault(require("./fontWeight"));
exports.fontWeight = fontWeight_1.default;
var sizes_1 = __importDefault(require("./sizes"));
exports.sizes = sizes_1.default;
var spacingMap_1 = __importDefault(require("./spacingMap"));
exports.spacingMap = spacingMap_1.default;
var tagMap_1 = __importDefault(require("./tagMap"));
exports.tagMap = tagMap_1.default;
var widthSizes_1 = __importDefault(require("./widthSizes"));
exports.widthSizes = widthSizes_1.default;
__exportStar(require("./marginInterface"), exports);
