// @frontend/wknd-components/dist/src/LegacyComponents/PageSlide/PageSlide.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePageSlideContext = exports.PageSlideContext = exports.onClickHandler = void 0;
var react_1 = __importStar(require("react"));
var react_dom_1 = require("react-dom");
var styled_components_1 = __importStar(require("styled-components"));
/** @deprecated this component has been deprecated */
var PageSlide = (0, react_1.createContext)({
    stack: [],
    setStack: function () { },
    handleOpenModal: function () { },
    handleCloseModal: function () { },
    handleCloseAllModals: function () { },
});
var emergence = (0, styled_components_1.keyframes)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tfrom {\n\t\ttransform: translateX(100%)\n\t}\n\n\tto {\n\t\ttransform: translateX(0)\n\t}\n"], ["\n\tfrom {\n\t\ttransform: translateX(100%)\n\t}\n\n\tto {\n\t\ttransform: translateX(0)\n\t}\n"])));
var closing = (0, styled_components_1.keyframes)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\tfrom {\n\t\ttransform: translateX(0)\n\t}\n\n\tto {\n\t\ttransform: translateX(100%)\n\t}\n"], ["\n\tfrom {\n\t\ttransform: translateX(0)\n\t}\n\n\tto {\n\t\ttransform: translateX(100%)\n\t}\n"])));
var background = (0, styled_components_1.keyframes)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tfrom {\n\t\tbackground-color: rgba(0, 0, 0, 0);\n\t}\n\n\tto {\n\t\tbackground-color: rgba(0, 0, 0, .2);\n\t}\n"], ["\n\tfrom {\n\t\tbackground-color: rgba(0, 0, 0, 0);\n\t}\n\n\tto {\n\t\tbackground-color: rgba(0, 0, 0, .2);\n\t}\n"])));
var hidingBackground = (0, styled_components_1.keyframes)(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tfrom {\n\t\tbackground-color: rgba(0, 0, 0, .2);\n\t}\n\n\tto {\n\t\tbackground-color: rgba(0, 0, 0, 0);\n\t}\n"], ["\n\tfrom {\n\t\tbackground-color: rgba(0, 0, 0, .2);\n\t}\n\n\tto {\n\t\tbackground-color: rgba(0, 0, 0, 0);\n\t}\n"])));
var SlideContainer = styled_components_1.default.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\twidth: calc(", "% - 110px);\n\theight: 100%;\n\toverflow-y: auto;\n\tbackground: white;\n\tborder-radius: 15px 0 0 15px;\n\tposition: absolute;\n\ttop: 0;\n\tright: 0;\n\tz-index: ", ";\n\tanimation: ", " .2s linear;\n"], ["\n\twidth: calc(", "% - 110px);\n\theight: 100%;\n\toverflow-y: auto;\n\tbackground: white;\n\tborder-radius: 15px 0 0 15px;\n\tposition: absolute;\n\ttop: 0;\n\tright: 0;\n\tz-index: ", ";\n\tanimation: ", " .2s linear;\n"])), function (_a) {
    var idx = _a.idx;
    return 100 - (idx * 2);
}, function (_a) {
    var idx = _a.idx;
    return idx;
}, function (props) { return props.closing ? closing : emergence; });
var LeftBackgroundCover = styled_components_1.default.div(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\tposition: absolute;\n\ttop: 0;\n\tleft: 0;\n\twidth: 100%;\n\theight: 100%;\n\tbackground-color: rgba(0, 0, 0, .2);\n\tz-index: ", ";\n\tanimation: ", " .2s linear;\n"], ["\n\tposition: absolute;\n\ttop: 0;\n\tleft: 0;\n\twidth: 100%;\n\theight: 100%;\n\tbackground-color: rgba(0, 0, 0, .2);\n\tz-index: ", ";\n\tanimation: ", " .2s linear;\n"])), function (props) { return props.idx; }, function (props) { return props.closing ? hidingBackground : background; });
var onClickHandler = function (closeClickOutside, onClose) {
    if (closeClickOutside === false) {
        return;
    }
    onClose();
};
exports.onClickHandler = onClickHandler;
var ModalContainer = function (_a) {
    var children = _a.children, closing = _a.closing, idx = _a.idx, onClose = _a.onClose, _b = _a.closeClickOutside, closeClickOutside = _b === void 0 ? true : _b;
    return ((0, react_dom_1.createPortal)(react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(LeftBackgroundCover, { closing: closing, idx: idx, onClick: function () { return (0, exports.onClickHandler)(closeClickOutside, onClose); } }),
        react_1.default.createElement(SlideContainer, { closing: closing, idx: idx }, children)), document.getElementById("root")));
};
/** @deprecated this component has been deprecated */
var PageSlideContext = function (_a) {
    var children = _a.children;
    var _b = (0, react_1.useState)([]), stack = _b[0], setStack = _b[1];
    var stackRef = (0, react_1.useRef)(null);
    (0, react_1.useEffect)(function () {
        stackRef.current = stack;
        if (stack.length) {
            document.body.style.overflow = "hidden";
        }
        else {
            document.body.style.overflow = "unset";
        }
        return function () {
            document.body.style.overflow = "unset";
        };
    }, [stack]);
    var handleOpenModal = function (element) { return setStack(function (prev) {
        var newStack = __spreadArray(__spreadArray([], prev, true), [element], false);
        if (newStack.length === 6) {
            return newStack.slice(1);
        }
        return newStack;
    }); };
    var handleCloseModal = function (id) {
        setStack(function (prev) { return prev.map(function (el) { return el.id === id ? __assign(__assign({}, el), { closing: true }) : el; }); });
        var timerId = setTimeout(function () {
            setStack(function (prev) { return prev.filter(function (el) { return el.id !== id; }); });
            clearTimeout(timerId);
        }, 180);
    };
    var handleCloseAllModals = function () {
        var stack = stackRef.current || [];
        var promiseArray = [];
        var _loop_1 = function (i) {
            var promiseItems = new Promise(function () {
                setStack(function (prev) { return prev.map(function (el) { return el.id === stack[i].id ? __assign(__assign({}, el), { closing: true }) : el; }); });
                setTimeout(function () {
                    setStack(function (prev) { return prev.filter(function (el) { return el.id !== stack[i].id; }); });
                }, 180);
            });
            promiseArray.push(promiseItems);
        };
        for (var i = stack.length - 1; i >= 0; i -= 1) {
            _loop_1(i);
        }
        Promise.all(promiseArray);
        stackRef.current = null;
    };
    return (react_1.default.createElement(PageSlide.Provider, { value: {
            stack: stack,
            setStack: setStack,
            handleOpenModal: handleOpenModal,
            handleCloseModal: handleCloseModal,
            handleCloseAllModals: handleCloseAllModals,
        } },
        children,
        stack.map(function (_a, i) {
            var id = _a.id, children = _a.children, closing = _a.closing;
            return (react_1.default.createElement(ModalContainer, { key: id, closing: closing, idx: i, onClose: function () { return handleCloseModal(id); }, closeClickOutside: false }, children));
        })));
};
exports.PageSlideContext = PageSlideContext;
/** @deprecated this component has been deprecated */
var usePageSlideContext = function () { return (0, react_1.useContext)(PageSlide); };
exports.usePageSlideContext = usePageSlideContext;
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6;
