// @frontend/wknd-components/dist/src/Pagination/Pagination.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var ButtonGroup_1 = require("../ButtonGroup");
var styleguide_1 = require("../styleguide");
var Typography_1 = __importDefault(require("../Typography/Typography"));
var RowsPerPageDropdown_1 = __importDefault(require("./RowsPerPageDropdown/RowsPerPageDropdown"));
var Pagination_styles_1 = require("./Pagination.styles");
var defaultRowOptions = [
    { value: 20, label: "20" },
    { value: 50, label: "50" },
    { value: 100, label: "100" }
];
var Pagination = function (_a) {
    var _b = _a.rowOptions, rowOptions = _b === void 0 ? defaultRowOptions : _b, _c = _a.rowsPerPage, rowsPerPage = _c === void 0 ? 20 : _c, rowCount = _a.rowCount, onChangePage = _a.onChangePage, onChangeRowsPerPage = _a.onChangeRowsPerPage, currentPage = _a.currentPage, side = _a.side, align = _a.align, dataQA = _a.dataQA, ariaLabel = _a.ariaLabel, className = _a.className, m = _a.m, ml = _a.ml, mr = _a.mr, mt = _a.mt, mb = _a.mb;
    var firstShownRow = currentPage === 1 ? 1 : (currentPage - 1) * rowsPerPage + 1;
    var lastShownRow = currentPage * rowsPerPage > rowCount ? rowCount : currentPage * rowsPerPage;
    var handlePreviousButtonClick = function () {
        onChangePage(currentPage - 1, rowCount);
    };
    var handleNextButtonClick = function () {
        onChangePage(currentPage + 1, rowCount);
    };
    var buttons = [
        {
            onClick: handlePreviousButtonClick,
            buttonText: "Previous",
            leftIcon: "ArrowLeft",
            dataQA: "".concat(dataQA, "-previous-button"),
            ariaLabel: "Previous Page",
            $isDisabled: currentPage === 1,
        },
        {
            onClick: handleNextButtonClick,
            buttonText: "Next",
            rightIcon: "ArrowRight",
            dataQA: "".concat(dataQA, "-next-button"),
            ariaLabel: "Next Page",
            $isDisabled: currentPage === Math.ceil(rowCount / rowsPerPage),
        },
    ];
    return (react_1.default.createElement(Pagination_styles_1.StyledPagination, { "data-qa": "".concat(dataQA, "-pagination"), className: className, m: m, ml: ml, mr: mr, mt: mt, mb: mb },
        rowOptions.length > 0 ?
            (react_1.default.createElement(Pagination_styles_1.PaginationContentWrapper, null,
                react_1.default.createElement(Typography_1.default, { variant: "bodyCopySmall", component: "span", dataQA: "items-per-page", color: "".concat(styleguide_1.colors.black, "A8"), m: "0 ".concat(styleguide_1.spacingMap.xxs, " 0 0") }, "Items per page"),
                react_1.default.createElement(RowsPerPageDropdown_1.default, { rowOptions: rowOptions, rowsPerPage: rowsPerPage, dataQA: dataQA, side: side, align: align, ariaLabel: ariaLabel, onChangeRowsPerPage: onChangeRowsPerPage, currentPage: currentPage }))) :
            null,
        react_1.default.createElement(Pagination_styles_1.PaginationContentWrapper, null,
            react_1.default.createElement(Typography_1.default, { variant: "bodyCopySmall", component: "span", dataQA: "items-shown", color: "".concat(styleguide_1.colors.black, "A8"), "aria-label": "Showing rows ".concat(firstShownRow, " to ").concat(lastShownRow, " of ").concat(rowCount, " total rows "), m: "0 ".concat(styleguide_1.spacingMap.xs, " 0 0") },
                "Showing ",
                firstShownRow,
                " - ",
                lastShownRow,
                " of ",
                rowCount),
            react_1.default.createElement(ButtonGroup_1.ButtonGroup, { buttonSize: "small", buttons: buttons, dataQA: "pagination-buttons", ariaLabel: "pagination buttons" }))));
};
Pagination.displayName = "Pagination";
exports.default = Pagination;
