// @frontend/wknd-components/dist/src/Pattern/Patterns/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Visionary = exports.Trailblazer = exports.Storyteller = exports.Sidekick = exports.Scientist = exports.Rebel = exports.Perfectionist = exports.Leader = exports.Innovator = exports.Helper = exports.Guru = exports.Entertainer = exports.Athlete = exports.Artist = exports.Adventurer = exports.Academic = void 0;
var Academic_1 = __importDefault(require("./Academic"));
exports.Academic = Academic_1.default;
var Adventurer_1 = __importDefault(require("./Adventurer"));
exports.Adventurer = Adventurer_1.default;
var Artist_1 = __importDefault(require("./Artist"));
exports.Artist = Artist_1.default;
var Athlete_1 = __importDefault(require("./Athlete"));
exports.Athlete = Athlete_1.default;
var Entertainer_1 = __importDefault(require("./Entertainer"));
exports.Entertainer = Entertainer_1.default;
var Guru_1 = __importDefault(require("./Guru"));
exports.Guru = Guru_1.default;
var Helper_1 = __importDefault(require("./Helper"));
exports.Helper = Helper_1.default;
var Innovator_1 = __importDefault(require("./Innovator"));
exports.Innovator = Innovator_1.default;
var Leader_1 = __importDefault(require("./Leader"));
exports.Leader = Leader_1.default;
var Perfectionist_1 = __importDefault(require("./Perfectionist"));
exports.Perfectionist = Perfectionist_1.default;
var Rebel_1 = __importDefault(require("./Rebel"));
exports.Rebel = Rebel_1.default;
var Scientist_1 = __importDefault(require("./Scientist"));
exports.Scientist = Scientist_1.default;
var Sidekick_1 = __importDefault(require("./Sidekick"));
exports.Sidekick = Sidekick_1.default;
var Storyteller_1 = __importDefault(require("./Storyteller"));
exports.Storyteller = Storyteller_1.default;
var Trailblazer_1 = __importDefault(require("./Trailblazer"));
exports.Trailblazer = Trailblazer_1.default;
var Visionary_1 = __importDefault(require("./Visionary"));
exports.Visionary = Visionary_1.default;
// List of Patterns numbers and names in figma:
// 16 = adventurer
// 04 = artist
// 07 = leader
// 08 = sidekick
// 09 = athlete
// 06 = storyteller
// 10 = visionary
// 05 = perfectionist
// 11 = helper
// 12 = rebel
// 13 = innovator
// 14 = guru
// 15 = academic
// 19 = entertainer
// 17 = scientist
// 18 = trailblazer
