// @frontend/wknd-components/dist/src/LegacyComponents/LegacyButton/legacyButtonVariants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.variants = void 0;
var styleguide_1 = require("../../styleguide");
exports.variants = {
    default: {
        color: styleguide_1.colors.white,
        iconColor: styleguide_1.colors.white,
        backgroundColor: styleguide_1.colors.midBlue,
        hover: styleguide_1.colors.midBlue,
        visited: styleguide_1.colors.midBlue,
        disabled: styleguide_1.colors.darkGray,
        disabledText: styleguide_1.colors.lightGray,
        border: "1px solid transparent",
    },
    green: {
        color: styleguide_1.colors.white,
        iconColor: styleguide_1.colors.white,
        backgroundColor: styleguide_1.colors.teal,
        hover: styleguide_1.colors.teal,
        visited: styleguide_1.colors.teal,
        disabled: styleguide_1.colors.darkGray,
        disabledText: styleguide_1.colors.lightGray,
        border: "1px solid transparent",
    },
    black: {
        color: styleguide_1.colors.white,
        iconColor: styleguide_1.colors.white,
        backgroundColor: styleguide_1.colors.black,
        hover: styleguide_1.colors.blackGray,
        visited: styleguide_1.colors.black,
        disabled: styleguide_1.colors.blackGray,
        disabledText: styleguide_1.colors.darkerGray,
        border: "1px solid transparent",
    },
    secondary: {
        color: styleguide_1.colors.black,
        iconColor: styleguide_1.colors.black,
        backgroundColor: "transparent",
        hover: styleguide_1.colors.black,
        hoverTextColor: styleguide_1.colors.white,
        visited: styleguide_1.colors.black,
        disabled: "transparent",
        disabledText: styleguide_1.colors.darkGray,
        disabledBorder: "1px solid ".concat(styleguide_1.colors.darkGray),
        border: "1px solid ".concat(styleguide_1.colors.black),
    },
    white: {
        color: styleguide_1.colors.black,
        iconColor: styleguide_1.colors.black,
        backgroundColor: styleguide_1.colors.white,
        disabled: styleguide_1.colors.darkGray,
        disabledText: styleguide_1.colors.lightGray,
        border: "1px solid transparent",
    },
    tertiary: {
        color: styleguide_1.colors.black,
        iconColor: styleguide_1.colors.black,
        backgroundColor: "transparent",
        disabled: styleguide_1.colors.darkGray,
        disabledText: styleguide_1.colors.lightGray,
        border: "none",
    },
    custom: {},
};
