// @frontend/wknd-components/dist/src/Drawer/Drawer.styles.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomDrawerStyles = exports.drawerWidth = void 0;
var styleguide_1 = require("../styleguide");
var rightPositionStyles = {
    top: styleguide_1.spacingMap.xs,
    bottom: styleguide_1.spacingMap.xs,
    right: styleguide_1.spacingMap.xs,
    left: "auto",
};
var drawerPosition = {
    right: rightPositionStyles,
};
exports.drawerWidth = {
    small: "480px",
    medium: "600px",
    large: "960px",
    // Adjust the Drawer width to fit the window, accounting for the default left inset set to auto.
    fullScreen: "calc(100% - 1rem)",
};
var getCustomDrawerStyles = function (_a) {
    var _b = _a.position, position = _b === void 0 ? "right" : _b, _c = _a.$zIndex, $zIndex = _c === void 0 ? 99 : _c, _d = _a.size, size = _d === void 0 ? "small" : _d;
    return ({
        overlay: {
            backgroundColor: "transparent",
            zIndex: $zIndex,
        },
        content: __assign({ width: exports.drawerWidth[size], borderRadius: "16px", border: "none", borderTop: "1px solid ".concat(styleguide_1.colors.white), backgroundColor: styleguide_1.colors.lightestGray, padding: 0, display: "flex", flexDirection: "column", boxShadow: "0px 2px 2px 0px rgba(0, 0, 0, 0.08),\n\t\t\t0px 4px 4px 0px rgba(0, 0, 0, 0.08), 0px 8px 16px -4px rgba(0, 0, 0, 0.08)", top: styleguide_1.spacingMap.xs, bottom: styleguide_1.spacingMap.xs, right: styleguide_1.spacingMap.xs, left: "auto" }, drawerPosition[position]),
    });
};
exports.getCustomDrawerStyles = getCustomDrawerStyles;
