// @frontend/wknd-components/dist/src/ProgressBar/ProgressBar.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyledWrapper = exports.ErrorText = exports.FullWidthRow = exports.StyledIconWrapper = exports.StyledProgressCount = exports.StyledProgressBar = void 0;
var styled_components_1 = __importStar(require("styled-components"));
var styleguide_1 = require("../styleguide");
var Typography_1 = __importDefault(require("../Typography/Typography"));
var BackgroundColors = {
    default: "transparent",
    error: styleguide_1.colors.red,
    complete: styleguide_1.colors.teal,
};
var getValueBackgroundImage = function (_a) {
    var state = _a.state, valueBarColor = _a.valueBarColor, valueBarPattern = _a.valueBarPattern;
    if (state !== "default" || valueBarColor) {
        return "";
    }
    if (valueBarPattern) {
        return "url(".concat(valueBarPattern, ")");
    }
    return "url(\"/assets/images/backgrounds/ProgressBarDefaultBackground.svg\")";
};
var getValueBackgroundColor = function (_a) {
    var state = _a.state, valueBarColor = _a.valueBarColor;
    if (state === "default" && valueBarColor) {
        return valueBarColor;
    }
    return BackgroundColors[state];
};
var progressValue = (0, styled_components_1.css)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tborder: none;\n\tborder-radius: ", ";\n\tbackground-color: ", ";\n\tbackground-image: ", ";\n\tbackground-repeat: repeat-x;\n\tbackground-size: auto;\n"], ["\n\tborder: none;\n\tborder-radius: ", ";\n\tbackground-color: ", ";\n\tbackground-image: ", ";\n\tbackground-repeat: repeat-x;\n\tbackground-size: auto;\n"])), function (_a) {
    var _b = _a.value, value = _b === void 0 ? 0 : _b, max = _a.max;
    return +value >= max ? "24px" : "24px 0 0 24px";
}, getValueBackgroundColor, getValueBackgroundImage);
exports.StyledProgressBar = styled_components_1.default.progress(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\theight: 9px;\n\tdisplay: flex;\n\tappearance: none;\n\tborder: 0;\n\tflex-grow: 3;\n\tflex-basis: 88%;\n\tborder-radius: 24px;\n\toverflow: hidden;\n\t::-webkit-progress-bar {\n\t\tbackground: ", ";\n\t\tborder-radius: 24px;\n\t}\n\t::-webkit-progress-value {\n\t\t", "\n\t}\n\t::-moz-progress-bar {\n\t\t", "\n\t}\n"], ["\n\theight: 9px;\n\tdisplay: flex;\n\tappearance: none;\n\tborder: 0;\n\tflex-grow: 3;\n\tflex-basis: 88%;\n\tborder-radius: 24px;\n\toverflow: hidden;\n\t::-webkit-progress-bar {\n\t\tbackground: ", ";\n\t\tborder-radius: 24px;\n\t}\n\t::-webkit-progress-value {\n\t\t", "\n\t}\n\t::-moz-progress-bar {\n\t\t", "\n\t}\n"])), function (_a) {
    var backgroundColor = _a.backgroundColor;
    return backgroundColor || styleguide_1.colors.white;
}, progressValue, progressValue);
exports.StyledProgressCount = (0, styled_components_1.default)(Typography_1.default)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tcolor: ", ";\n\tmargin-left: 1.25rem;\n\twidth: 24px;\n"], ["\n\tcolor: ", ";\n\tmargin-left: 1.25rem;\n\twidth: 24px;\n"])), styleguide_1.colors.black);
exports.StyledIconWrapper = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tdisplay: flex;\n\tflex: 1;\n\tmargin-left: ", ";\n\tjustify-content: space-between;\n\tmax-width: fit-content;\n"], ["\n\tdisplay: flex;\n\tflex: 1;\n\tmargin-left: ", ";\n\tjustify-content: space-between;\n\tmax-width: fit-content;\n"])), styleguide_1.spacingMap.md);
exports.FullWidthRow = styled_components_1.default.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\tdisplay: flex;\n\twidth: 100%;\n\talign-items: center;\n\tmargin-bottom: 0.313rem;\n\tjustify-content: space-between;\n"], ["\n\tdisplay: flex;\n\twidth: 100%;\n\talign-items: center;\n\tmargin-bottom: 0.313rem;\n\tjustify-content: space-between;\n"])));
exports.ErrorText = (0, styled_components_1.default)(Typography_1.default)(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\tdisplay: flex;\n\tmax-width: 100%;\n\tflex: 1 10%;\n"], ["\n\tdisplay: flex;\n\tmax-width: 100%;\n\tflex: 1 10%;\n"])));
exports.StyledWrapper = styled_components_1.default.div(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n\tdisplay: flex;\n\tflex-flow: row wrap;\n\talign-items: center;\n\tjustify-content: flex-start;\n\tflex-wrap: wrap;\n\twidth: 100%;\n\t", "\n"], ["\n\tdisplay: flex;\n\tflex-flow: row wrap;\n\talign-items: center;\n\tjustify-content: flex-start;\n\tflex-wrap: wrap;\n\twidth: 100%;\n\t", "\n"])), styleguide_1.marginsCss);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7;
