// @frontend/wknd-components/dist/src/LegacyComponents/TextArea/TextArea.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorMessageSpan = exports.Message = exports.StyledTextArea = exports.ContainerTextArea = exports.LabelWrapper = exports.LINE_HEIGHT = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var styleguide_1 = require("../../styleguide");
exports.LINE_HEIGHT = 15;
exports.LabelWrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tfont-size: ", ";\n\tline-height: 15px;\n\tfont-family: ", ";\n\tcolor: ", ";\n"], ["\n\tfont-size: ", ";\n\tline-height: 15px;\n\tfont-family: ", ";\n\tcolor: ", ";\n"])), styleguide_1.fontSizes.s, styleguide_1.fontFamily.larsseit, function (props) {
    if (props.$isInvalid) {
        return styleguide_1.colors.red;
    }
    return styleguide_1.colors.black;
});
exports.ContainerTextArea = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\twidth: ", ";\n"], ["\n\twidth: ", ";\n"])), function (_a) {
    var widthSize = _a.widthSize;
    return widthSize;
});
exports.StyledTextArea = styled_components_1.default.textarea(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tresize: ", ";\n\tfont-family: ", ";\n\tfont-size: ", ";\n\tborder: 1px solid ", ";\n\tmargin-top: 5px;\n\tpadding: 10px 8px 10px 16px;\n\twidth: ", ";\n\tbackground: ", ";\n\tbox-sizing: border-box;\n\tborder-radius: 4px;\n\tline-height: ", "px;\n\n\t&:focus {\n\t\toutline-offset: 2px;\n\t\toutline: 3px solid ", "33;\n\t}\n\n\tborder-color: ", ";\n"], ["\n\tresize: ", ";\n\tfont-family: ", ";\n\tfont-size: ", ";\n\tborder: 1px solid ", ";\n\tmargin-top: 5px;\n\tpadding: 10px 8px 10px 16px;\n\twidth: ", ";\n\tbackground: ", ";\n\tbox-sizing: border-box;\n\tborder-radius: 4px;\n\tline-height: ", "px;\n\n\t&:focus {\n\t\toutline-offset: 2px;\n\t\toutline: 3px solid ", "33;\n\t}\n\n\tborder-color: ", ";\n"])), function (_a) {
    var resize = _a.resize;
    return resize;
}, styleguide_1.fontFamily.larsseit, styleguide_1.fontSizes.s, styleguide_1.colors.defaultGray, function (_a) {
    var widthSize = _a.widthSize;
    return widthSize;
}, styleguide_1.colors.white, exports.LINE_HEIGHT, styleguide_1.colors.midBlue, function (props) {
    if (props.$isDisabled) {
        return styleguide_1.colors.defaultGray;
    }
    if (props.$isInvalid) {
        return styleguide_1.colors.red;
    }
    return styleguide_1.colors.defaultGray;
});
exports.Message = styled_components_1.default.p(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tfont-family: ", ";\n\ttext-align: ", ";\n\tcolor: ", ";\n\tfont-size: 12px;\n\tmargin-top: 10px;\n\tmargin-bottom: 0;\n"], ["\n\tfont-family: ", ";\n\ttext-align: ", ";\n\tcolor: ", ";\n\tfont-size: 12px;\n\tmargin-top: 10px;\n\tmargin-bottom: 0;\n"])), styleguide_1.fontFamily.larsseit, function (_a) {
    var positionErrorMessage = _a.positionErrorMessage;
    return positionErrorMessage;
}, styleguide_1.colors.darkerGray);
exports.ErrorMessageSpan = styled_components_1.default.span(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\tcolor: ", "\n"], ["\n\tcolor: ", "\n"])), styleguide_1.colors.red);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
