// @frontend/wknd-components/dist/src/Spinner/Spinner.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Svg = exports.Circle = exports.StyledSpinner = exports.spinnerSize = void 0;
var styled_components_1 = __importStar(require("styled-components"));
var styleguide_1 = require("../styleguide");
exports.spinnerSize = {
    Small: "40px",
    Medium: "72px",
    Large: "96px",
    XLarge: "160px",
};
var loadingAnimationStrokeWidth = (0, styled_components_1.keyframes)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    0% {\n        stroke-width: 10;\n    }\n\n    50% {\n        stroke-width: 1;\n    }\n\n    100% {\n        stroke-width: 3;\n    }\n"], ["\n    0% {\n        stroke-width: 10;\n    }\n\n    50% {\n        stroke-width: 1;\n    }\n\n    100% {\n        stroke-width: 3;\n    }\n"])));
var loadingAnimationStrokeDash = (0, styled_components_1.keyframes)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    0% {\n      stroke-dasharray: 1, 200;\n      stroke-dashoffset: 0;\n    }\n\n    50% {\n      stroke-dasharray: 89, 200;\n      stroke-dashoffset: -35px;\n    }\n\n    100% {\n      stroke-dasharray: 89, 200;\n      stroke-dashoffset: -124px;\n    }\n"], ["\n    0% {\n      stroke-dasharray: 1, 200;\n      stroke-dashoffset: 0;\n    }\n\n    50% {\n      stroke-dasharray: 89, 200;\n      stroke-dashoffset: -35px;\n    }\n\n    100% {\n      stroke-dasharray: 89, 200;\n      stroke-dashoffset: -124px;\n    }\n"])));
var loadingAnimationStrokeColor = (0, styled_components_1.keyframes)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\t0%,\n\t100% {\n        stroke: ", ";\n    }\n"], ["\n\t0%,\n\t100% {\n        stroke: ", ";\n    }\n"])), styleguide_1.colors.midBlue);
var loadingAnimationRotate = (0, styled_components_1.keyframes)(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n    100% {\n      transform: rotate(360deg);\n    }\n"], ["\n    100% {\n      transform: rotate(360deg);\n    }\n"])));
exports.StyledSpinner = styled_components_1.default.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n    position: relative;\n    pointer-events: none;\n    width: ", ";\n    height: ", ";\n"], ["\n    position: relative;\n    pointer-events: none;\n    width: ", ";\n    height: ", ";\n"])), function (_a) {
    var size = _a.size;
    return size && exports.spinnerSize[size];
}, function (_a) {
    var size = _a.size;
    return size && exports.spinnerSize[size];
});
exports.Circle = styled_components_1.default.circle(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n    stroke-width: 3;\n    stroke-linecap: round;\n\tanimation: ", " .5s ease-in-out,\n\t\t", " 1.5s ease-in-out infinite,\n\t\t", " 6s ease-in-out infinite;\n"], ["\n    stroke-width: 3;\n    stroke-linecap: round;\n\tanimation: ", " .5s ease-in-out,\n\t\t", " 1.5s ease-in-out infinite,\n\t\t", " 6s ease-in-out infinite;\n"])), loadingAnimationStrokeWidth, loadingAnimationStrokeDash, loadingAnimationStrokeColor);
exports.Svg = styled_components_1.default.svg(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n    animation: ", " 2s linear infinite;\n"], ["\n    animation: ", " 2s linear infinite;\n"])), loadingAnimationRotate);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7;
