// @frontend/wknd-components/dist/src/Modal/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModalImage = exports.ModalHeading = exports.ModalHeader = exports.ModalFooter = exports.ModalBody = exports.Modal = void 0;
var Modal_1 = __importDefault(require("./Modal"));
exports.Modal = Modal_1.default;
var ModalBody_1 = __importDefault(require("./ModalBody"));
exports.ModalBody = ModalBody_1.default;
var ModalFooter_1 = require("./ModalFooter");
Object.defineProperty(exports, "ModalFooter", { enumerable: true, get: function () { return ModalFooter_1.ModalFooter; } });
var ModalHeader_1 = __importDefault(require("./ModalHeader"));
exports.ModalHeader = ModalHeader_1.default;
var ModalHeading_1 = __importDefault(require("./ModalHeading"));
exports.ModalHeading = ModalHeading_1.default;
var ModalImage_1 = __importDefault(require("./ModalImage"));
exports.ModalImage = ModalImage_1.default;
__exportStar(require("./Modal.types"), exports);
