// @frontend/wknd-components/dist/src/Modal/Modal.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomModalStyles = exports.modalWidth = void 0;
var styleguide_1 = require("../styleguide");
exports.modalWidth = {
    small: "400px",
    medium: "512px",
    large: "752px",
    extra_large: "1080px",
    fullScreen: "100%",
};
var getModalWidth = function (variant, size) {
    if (variant === "destructive") {
        return exports.modalWidth.small;
    }
    if (variant === "scrollable") {
        return exports.modalWidth.medium;
    }
    if (variant === "graphic") {
        return exports.modalWidth.small;
    }
    if (variant === "graphic_large") {
        return exports.modalWidth.medium;
    }
    return size && exports.modalWidth[size];
};
var getCustomModalStyles = function (_a) {
    var size = _a.size, variant = _a.variant;
    return ({
        overlay: {
            backgroundColor: "transparent",
            zIndex: 9999,
        },
        content: {
            width: variant && getModalWidth(variant, size),
            backgroundColor: (variant === null || variant === void 0 ? void 0 : variant.includes("graphic")) ? styleguide_1.colors.black : styleguide_1.colors.white,
            padding: "0",
            display: "flex",
            flexDirection: "column",
            height: size === "fullScreen" ? "100%" : "fit-content",
            position: "fixed",
            top: "50%",
            left: "50%",
            transform: "translate(-50%, -50%)",
            // eslint-disable-next-line max-len
            boxShadow: "0px 60px 240px rgba(0, 0, 0, 0.5), 0px 32px 32px -8px rgba(0, 0, 0, 0.08), 0px 16px 16px rgba(0, 0, 0, 0.08), 0px 8px 8px rgba(0, 0, 0, 0.08), 0px 4px 4px rgba(0, 0, 0, 0.08), 0px 2px 2px rgba(0, 0, 0, 0.08)",
            borderRadius: size !== "fullScreen" ? "16px" : "0",
            border: "none",
        },
    });
};
exports.getCustomModalStyles = getCustomModalStyles;
