// @frontend/wknd-components/dist/src/Badge/Badge.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyledBadgeText = exports.StyledBadge = exports.BadgeVariantBackgroundColors = exports.BadgeVariantIconColors = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var styleguide_1 = require("../styleguide");
var Typography_1 = require("../Typography");
var utils_1 = require("../utils");
exports.BadgeVariantIconColors = {
    default: "".concat(styleguide_1.colors.darkerGray),
    active: "".concat(styleguide_1.colors.teal),
    testing: "".concat(styleguide_1.colors.gold),
    blue: "".concat(styleguide_1.colors.midBlue),
    error: "".concat(styleguide_1.colors.red),
    warning: "".concat(styleguide_1.colors.yellow),
};
exports.BadgeVariantBackgroundColors = {
    default: "".concat(styleguide_1.colors.lightGray),
    active: "".concat(styleguide_1.colors.teal, "40"),
    testing: "".concat(styleguide_1.colors.gold, "40"),
    blue: "".concat(styleguide_1.colors.midBlue, "40"),
    error: "".concat(styleguide_1.colors.red, "40"),
    warning: "".concat(styleguide_1.colors.yellow, "40"),
};
exports.StyledBadge = styled_components_1.default.span(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tdisplay: inline-flex;\n\tflex-direction: row;\n\talign-items: center;\n\tborder-radius: 4px;\n\twidth: fit-content;\n\theight: ", ";\n\tbackground-color: ", ";\n\t", ";\n"], ["\n\tdisplay: inline-flex;\n\tflex-direction: row;\n\talign-items: center;\n\tborder-radius: 4px;\n\twidth: fit-content;\n\theight: ", ";\n\tbackground-color: ", ";\n\t", ";\n"])), function (props) { return props.size === "small" ? "20px" : "24px"; }, function (props) { return props.variant && exports.BadgeVariantBackgroundColors[props.variant]; }, styleguide_1.marginsCss);
exports.StyledBadgeText = (0, styled_components_1.default)(Typography_1.Typography)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\ttext-wrap: nowrap;\n\tpadding: ", ";\n"], ["\n\ttext-wrap: nowrap;\n\tpadding: ", ";\n"])), function (props) { return props.size === "small" ? "0 ".concat(styleguide_1.spacingMap.xxs) : "".concat((0, utils_1.pxToRem)(2), " ").concat(styleguide_1.spacingMap.xs); });
var templateObject_1, templateObject_2;
