// @frontend/wknd-components/dist/src/Avatar/Avatar.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentWrapper = exports.StyledAvatar = exports.avatarVariantOptions = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var styleguide_1 = require("../styleguide");
exports.avatarVariantOptions = {
    Small: {
        size: "24px",
        iconSize: 16,
        copy: "caption",
    },
    Medium: {
        size: "32px",
        iconSize: 20,
        copy: "bodyCopyBold",
    },
    Large: {
        size: "40px",
        iconSize: 20,
        copy: "bodyCopyBold",
    },
};
exports.StyledAvatar = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tborder-radius: 100px;\n\t", "\n\tbackground-color: ", ";\n\toverflow: hidden;\n\n\t", ";\n"], ["\n\tborder-radius: 100px;\n\t", "\n\tbackground-color: ", ";\n\toverflow: hidden;\n\n\t", ";\n"])), function (_a) {
    var size = _a.size;
    return size && "\n\t\twidth: ".concat(exports.avatarVariantOptions[size].size, ";\n\t\theight: ").concat(exports.avatarVariantOptions[size].size, ";\n\t");
}, function (_a) {
    var backgroundColor = _a.backgroundColor;
    return backgroundColor || styleguide_1.colors.lightGray;
}, styleguide_1.marginsCss);
exports.ContentWrapper = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\theight: 100%;\n\tdisplay: flex;\n\tjustify-content: center;\n\toverflow: hidden;\n\talign-items: center;\n\t& > img {\n\t\tborder-radius: 100px;\n\t\twidth: 100%;\n   \t\theight: 100%;\n\t}\n"], ["\n\theight: 100%;\n\tdisplay: flex;\n\tjustify-content: center;\n\toverflow: hidden;\n\talign-items: center;\n\t& > img {\n\t\tborder-radius: 100px;\n\t\twidth: 100%;\n   \t\theight: 100%;\n\t}\n"])));
var templateObject_1, templateObject_2;
