// @frontend/wknd-components/dist/src/FlowCard/FlowCard.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowCardIcon = exports.FlowCardHeaderWrapper = exports.StyledFlowCard = exports.iconBackgroundColors = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var styleguide_1 = require("../styleguide");
exports.iconBackgroundColors = {
    primary: styleguide_1.colors.darkBlue,
    success: styleguide_1.colors.teal,
    default: styleguide_1.colors.gold,
};
exports.StyledFlowCard = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tdisplay: flex;\n\tflex-direction: column;\n\tjustify-content: flex-start;\n\talign-items: flex-start;\n\twidth: ", ";\n\theight: 100%;\n\tbackground: ", ";\n\tbox-shadow: 0 0 1px ", "52;\n\tborder-radius: 16px;\n\tpadding: ", ";\n\toverflow: hidden;\n\n\tborder: ", ";\n\n\t&:hover {\n\t\toutline: ", ";\n\t\tcursor: pointer;\n\t}\n\n\t&:active {\n\t\tbox-shadow: 0 2px 16px -2px ", "40;\n\t\toutline: ", ";\n\t}\n"], ["\n\tdisplay: flex;\n\tflex-direction: column;\n\tjustify-content: flex-start;\n\talign-items: flex-start;\n\twidth: ", ";\n\theight: 100%;\n\tbackground: ", ";\n\tbox-shadow: 0 0 1px ", "52;\n\tborder-radius: 16px;\n\tpadding: ", ";\n\toverflow: hidden;\n\n\tborder: ", ";\n\n\t&:hover {\n\t\toutline: ", ";\n\t\tcursor: pointer;\n\t}\n\n\t&:active {\n\t\tbox-shadow: 0 2px 16px -2px ", "40;\n\t\toutline: ", ";\n\t}\n"])), function (_a) {
    var maxCardWidth = _a.maxCardWidth;
    return maxCardWidth || "225px";
}, styleguide_1.colors.white, styleguide_1.colors.black, styleguide_1.spacingMap.xs, function (_a) {
    var isInvalid = _a.isInvalid;
    return isInvalid ? "1px ".concat(styleguide_1.colors.destructive, " solid") : "none";
}, function (_a) {
    var isInvalid = _a.isInvalid;
    return !isInvalid && "1px ".concat(styleguide_1.colors.black, " solid");
}, styleguide_1.colors.midBlue, function (_a) {
    var isInvalid = _a.isInvalid;
    return !isInvalid && "2px ".concat(styleguide_1.colors.midBlue, " solid");
});
exports.FlowCardHeaderWrapper = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\tdisplay: flex;\n\talign-items: center;\n\twidth: 100%;\n"], ["\n\tdisplay: flex;\n\talign-items: center;\n\twidth: 100%;\n"])));
exports.FlowCardIcon = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tborder-radius: 8px;\n\tpadding: ", ";\n\theight: 16px;\n\twidth: 16px;\n\tbackground-color: ", "\n"], ["\n\tborder-radius: 8px;\n\tpadding: ", ";\n\theight: 16px;\n\twidth: 16px;\n\tbackground-color: ", "\n"])), styleguide_1.spacingMap.xs, function (_a) {
    var iconVariant = _a.iconVariant;
    return iconVariant && exports.iconBackgroundColors[iconVariant];
});
var templateObject_1, templateObject_2, templateObject_3;
