// @frontend/wknd-components/dist/src/DataTable/DataTable.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomDashboardText = exports.NoDataCustomWrapper = exports.IconWrapper = exports.customStyles = exports.TableWrapper = exports.HEAD_ROW_HEIGHT = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var styleguide_1 = require("../styleguide");
var Typography_1 = require("../Typography");
exports.HEAD_ROW_HEIGHT = "48px";
exports.TableWrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tposition: relative;\n\tbox-shadow: 0px 0px 1px 0px ", "52, 0px 2px 8px -2px ", "1a;\n\tbox-sizing: border-box;\n\tborder-radius: 8px;\n\tbackground: ", ";\n\t", ";\n\n\t", ";\n"], ["\n\tposition: relative;\n\tbox-shadow: 0px 0px 1px 0px ", "52, 0px 2px 8px -2px ", "1a;\n\tbox-sizing: border-box;\n\tborder-radius: 8px;\n\tbackground: ", ";\n\t", ";\n\n\t", ";\n"])), styleguide_1.colors.black, styleguide_1.colors.black, styleguide_1.colors.white, function (_a) {
    var hasPagination = _a.hasPagination;
    return hasPagination && "div:first-of-type {\n\t\tborder-bottom-left-radius: unset;\n\t\tborder-bottom-right-radius: unset; }";
}, styleguide_1.marginsCss);
var customStyles = function (rowHeight, expandableRow, selectedRowCount) { return ({
    table: {
        style: {
            fontFamily: styleguide_1.displayFontSizes.bodyCopyBold.$fontFamily,
        },
    },
    headRow: {
        style: {
            borderBottom: "1px solid ".concat(styleguide_1.colors.lightGray),
            minHeight: "initial",
            height: exports.HEAD_ROW_HEIGHT,
            boxSizing: "border-box",
            backgroundColor: selectedRowCount > 0 ? styleguide_1.colors.lightestGray : styleguide_1.colors.white,
        },
    },
    headCells: {
        style: {
            padding: styleguide_1.spacingMap.sm,
            "&>.rdt_TableCol_Sortable": {
                visibility: selectedRowCount > 0 ? "hidden" : "visible",
            },
            "&>div": {
                display: "flex",
                color: "".concat(styleguide_1.colors.black, "A8"),
                fontWeight: styleguide_1.displayFontSizes.bodyCopyBold.$fontWeight,
                fontSize: styleguide_1.displayFontSizes.bodyCopyBold.$fontSize,
                lineHeight: styleguide_1.displayFontSizes.bodyCopyBold.lineHeight,
                "&:focus": {
                    outlineOffset: "2px",
                    outline: "3px solid ".concat(styleguide_1.colors.midBlue, "33"),
                    color: styleguide_1.colors.black,
                },
                "&:active": {
                    color: styleguide_1.colors.black,
                },
            },
        },
    },
    rows: {
        style: __assign({ minHeight: "60px", "&:last-child": {
                borderBottom: "solid 1px ".concat(styleguide_1.colors.lightGray),
            } }, (expandableRow && {
            "&:hover": {
                backgroundColor: styleguide_1.colors.lightestGray,
            },
        })),
    },
    cells: {
        style: {
            padding: "0 ".concat(styleguide_1.spacingMap.sm),
            // target the inner div of the cell so that the width is more flexible
            "& > div[data-tag=allowRowEvents]": {
                height: rowHeight,
                display: "flex",
                alignItems: "center",
            },
        },
    },
    pagination: {
        style: {
            fontFamily: styleguide_1.fontFamily.larsseit,
            borderTop: "none",
            select: {
                "&:focus": {
                    outlineOffset: "2px",
                    outline: "3px solid ".concat(styleguide_1.colors.midBlue, "33"),
                },
            },
        },
        pageButtonsStyle: {
            "&:focus": {
                outlineOffset: "2px",
                outline: "3px solid ".concat(styleguide_1.colors.midBlue, "33"),
            },
        },
    },
    noData: {
        style: {
            h2: {
                fontSize: styleguide_1.displayFontSizes.displayMedium.$fontSize,
                fontFamily: styleguide_1.displayFontSizes.displayMedium.$fontFamily,
                fontWeight: styleguide_1.displayFontSizes.displayMedium.$fontWeight,
            },
            p: {
                width: styleguide_1.widthSizes.m,
                height: "40px",
                fontFamily: styleguide_1.displayFontSizes.bodyCopy.$fontFamily,
                fontSize: styleguide_1.displayFontSizes.bodyCopy.$fontSize,
                color: "".concat(styleguide_1.colors.black, "a8"),
                fontWeight: styleguide_1.displayFontSizes.bodyCopy.$fontWeight,
            },
            textAlign: "center",
            maxHeight: "700px",
        },
    },
    subHeader: {
        style: {
            borderTopLeftRadius: "8px",
            borderTopRightRadius: "8px",
            borderBottom: "1px solid ".concat(styleguide_1.colors.lightGray),
            borderTop: "solid 1px ".concat(styleguide_1.colors.lightGray),
            alignContent: "revert",
            display: "flex",
            justifyContent: "flex-start",
            padding: "".concat(styleguide_1.spacingMap.xs, " ").concat(styleguide_1.spacingMap.sm),
        },
    },
    responsiveWrapper: {
        style: {
            overflowY: "initial",
        },
    },
    expanderButton: {
        style: {
            ":hover:not(:disabled)": {
                backgroundColor: "transparent",
            },
            ":focus": {
                backgroundColor: "transparent",
            },
        },
    },
}); };
exports.customStyles = customStyles;
exports.IconWrapper = styled_components_1.default.button(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\twidth: auto;\n\theight: auto;\n\tmax-height: 18.8px;\n\tmargin: 6px 6.4px;\n\tbackground: transparent;\n\tborder: none;\n"], ["\n\twidth: auto;\n\theight: auto;\n\tmax-height: 18.8px;\n\tmargin: 6px 6.4px;\n\tbackground: transparent;\n\tborder: none;\n"])));
exports.NoDataCustomWrapper = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tpadding: ", " ", ";\n\tdisplay: flex;\n\tflex-direction: column;\n\tjustify-content: center;\n\talign-items: center;\n"], ["\n\tpadding: ", " ", ";\n\tdisplay: flex;\n\tflex-direction: column;\n\tjustify-content: center;\n\talign-items: center;\n"])), styleguide_1.spacingMap.lg, styleguide_1.spacingMap.sm);
exports.CustomDashboardText = (0, styled_components_1.default)(Typography_1.Typography)(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\theight: 100% !important;\n"], ["\n\theight: 100% !important;\n"])));
var templateObject_1, templateObject_2, templateObject_3, templateObject_4;
