// @frontend/wknd-components/dist/src/DataTable/DataTable.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_data_table_component_1 = __importDefault(require("react-data-table-component"));
var react_dom_1 = require("react-dom");
var Checkbox_1 = __importDefault(require("../Checkbox/Checkbox"));
var Icon_1 = require("../Icon/Icon");
var Pagination_1 = __importDefault(require("../Pagination/Pagination"));
var Spinner_1 = __importDefault(require("../Spinner/Spinner"));
var styleguide_1 = require("../styleguide");
var Typography_1 = __importDefault(require("../Typography/Typography"));
var DataTable_styles_1 = require("./DataTable.styles");
var TableActionsBar_1 = require("./TableActionsBar");
var NoDataCustom = function (_a) {
    var header = _a.header, message = _a.message, action = _a.action, image = _a.image;
    return (react_1.default.createElement(DataTable_styles_1.NoDataCustomWrapper, null,
        image,
        react_1.default.createElement(Typography_1.default, { variant: "displayMedium", component: "h2", dataQA: "dashboard-header", m: "".concat(styleguide_1.spacingMap.md, " 0 0 0") }, header),
        react_1.default.createElement(DataTable_styles_1.CustomDashboardText, { variant: "bodyCopy", component: "p", dataQA: "dashboard-message", m: "".concat(styleguide_1.spacingMap.xs, " 0 ").concat(styleguide_1.spacingMap.md, " 0") }, message),
        action));
};
// eslint-disable-next-line @typescript-eslint/comma-dangle, comma-spacing
var DataTable = function (_a) {
    var _b, _c;
    var dataQA = _a.dataQA, _d = _a.hasPagination, hasPagination = _d === void 0 ? false : _d, _e = _a.isLoading, isLoading = _e === void 0 ? false : _e, _f = _a.noDataHeader, noDataHeader = _f === void 0 ? "No records to show" : _f, _g = _a.noDataText, noDataText = _g === void 0 ? "" : _g, _h = _a.noDataImage, noDataImage = _h === void 0 ? null : _h, _j = _a.noDataAction, noDataAction = _j === void 0 ? null : _j, _k = _a.areRowsSelectable, areRowsSelectable = _k === void 0 ? false : _k, _l = _a.rowSpacing, rowSpacing = _l === void 0 ? "default" : _l, tableActionsProps = _a.tableActionsProps, onSelectedRowsChange = _a.onSelectedRowsChange, clearSelectedRows = _a.clearSelectedRows, subHeaderComponent = _a.subHeaderComponent, m = _a.m, mt = _a.mt, mb = _a.mb, ml = _a.ml, mr = _a.mr, props = __rest(_a, ["dataQA", "hasPagination", "isLoading", "noDataHeader", "noDataText", "noDataImage", "noDataAction", "areRowsSelectable", "rowSpacing", "tableActionsProps", "onSelectedRowsChange", "clearSelectedRows", "subHeaderComponent", "m", "mt", "mb", "ml", "mr"]);
    var defaultSortMessage = "not sorted yet";
    var _m = (0, react_1.useState)(defaultSortMessage), sortDirection = _m[0], setSortDirection = _m[1];
    var customSort = function (selectedColumn, direction) {
        var ariaDirection = direction === "desc" ? "down" : "up";
        var columnName = selectedColumn === null || selectedColumn === void 0 ? void 0 : selectedColumn.name;
        setSortDirection(columnName ? "sorted ".concat(ariaDirection, " by ").concat(columnName) : defaultSortMessage);
    };
    var getRowHeight = function () {
        switch (rowSpacing) {
            case "snug":
                return styleguide_1.spacingMap.xl;
            case "loose":
                return "4.5rem";
            default:
                return styleguide_1.spacingMap.xxl;
        }
    };
    var _o = (0, react_1.useState)(0), selectedCount = _o[0], setSelectedCount = _o[1];
    var _p = (0, react_1.useState)(false), toggleCleared = _p[0], setToggleCleared = _p[1];
    var tableWrapperRef = (0, react_1.useRef)(null);
    var setSortableColumnTabIndex = function (tabIndex) {
        var _a;
        var allSortableColumns = (_a = tableWrapperRef.current) === null || _a === void 0 ? void 0 : _a.querySelectorAll(".rdt_TableCol_Sortable");
        if (tabIndex === "0") {
            return allSortableColumns === null || allSortableColumns === void 0 ? void 0 : allSortableColumns.forEach(function (element) { return element.setAttribute("tabindex", "0"); });
        }
        allSortableColumns === null || allSortableColumns === void 0 ? void 0 : allSortableColumns.forEach(function (element) { return element.setAttribute("tabindex", "-1"); });
    };
    var handleDeselect = (0, react_1.useCallback)(function () {
        setToggleCleared(!toggleCleared);
        setSelectedCount(0);
        setSortableColumnTabIndex("0");
        if (tableActionsProps === null || tableActionsProps === void 0 ? void 0 : tableActionsProps.handleDeselect) {
            tableActionsProps.handleDeselect();
        }
    }, [toggleCleared, tableActionsProps]);
    // use clearSelectedRows in case it is passed by the parent application
    var handleClearSelectedRows = function () { return clearSelectedRows || toggleCleared; };
    var handleOnSelectedRowsChange = (0, react_1.useCallback)(function (selected) {
        if (onSelectedRowsChange) {
            // still call onSelectedRowsChange in case if a function is passed
            // so data can be returned to parent app
            onSelectedRowsChange(selected);
        }
        if (!tableActionsProps) {
            return;
        }
        if (selected.selectedCount === 0) {
            setSortableColumnTabIndex("0");
            setSelectedCount(0);
            return;
        }
        // set tabindex on columns to prevent tabbing when the action bar is showing
        setSortableColumnTabIndex("-1");
        setSelectedCount(selected.selectedCount);
    }, [onSelectedRowsChange, tableActionsProps]);
    (0, react_1.useEffect)(function () {
        if (clearSelectedRows) {
            handleOnSelectedRowsChange({
                allSelected: false,
                selectedCount: 0,
                selectedRows: [],
            });
        }
    }, [clearSelectedRows, handleOnSelectedRowsChange]);
    var selectableRowsComponentProps = {
        dataQA: "".concat(dataQA, "-checkbox"),
        isIndeterminate: function (isIndeterminate) { return isIndeterminate; },
    };
    var tableHead = (_b = tableWrapperRef.current) === null || _b === void 0 ? void 0 : _b.querySelector(".rdt_TableHead");
    return (react_1.default.createElement(DataTable_styles_1.TableWrapper, { ref: tableWrapperRef, "data-qa": dataQA, "aria-label": "Sortable Table", hasPagination: hasPagination, m: m, mt: mt, mb: mb, ml: ml, mr: mr },
        (selectedCount > 0 && tableHead && tableActionsProps) && ((0, react_dom_1.createPortal)(react_1.default.createElement(TableActionsBar_1.TableActionsBar, { selectedCountMessage: tableActionsProps.selectedCountMessage, handleDeselect: handleDeselect, tableBulkActions: tableActionsProps.tableBulkActions || [] }), tableHead)),
        react_1.default.createElement(react_data_table_component_1.default, __assign({ defaultSortAsc: false, customStyles: (0, DataTable_styles_1.customStyles)(getRowHeight(), (props === null || props === void 0 ? void 0 : props.expandableRows) || false, selectedCount), sortIcon: (react_1.default.createElement(DataTable_styles_1.IconWrapper, { "aria-label": sortDirection, tabIndex: -1 },
                react_1.default.createElement(Icon_1.Icon, { color: styleguide_1.colors.black, name: "ChevronDown" }))), persistTableHead: true, noDataComponent: isLoading
                ? (react_1.default.createElement(Spinner_1.default, { size: "Medium", dataQA: "Loader" }))
                : react_1.default.createElement(NoDataCustom, { header: noDataHeader, message: noDataText, action: noDataAction, image: noDataImage }), paginationPerPage: (_c = props.paginationPerPage) !== null && _c !== void 0 ? _c : 20, pagination: hasPagination, paginationComponent: Pagination_1.default, selectableRows: areRowsSelectable, selectableRowsComponent: Checkbox_1.default, onSort: customSort, selectableRowsComponentProps: selectableRowsComponentProps, expandableIcon: {
                collapsed: react_1.default.createElement(Icon_1.Icon, { name: "ChevronRight" }),
                expanded: react_1.default.createElement(Icon_1.Icon, { name: "ChevronDown" }),
            }, onSelectedRowsChange: handleOnSelectedRowsChange, clearSelectedRows: handleClearSelectedRows(), subHeaderComponent: subHeaderComponent, subHeader: !!subHeaderComponent }, props))));
};
DataTable.displayName = "DataTable";
exports.default = DataTable;
