// @frontend/wknd-components/dist/src/Combobox/Combobox.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var react_select_1 = __importDefault(require("react-select"));
var Label_1 = __importDefault(require("../Label/Label"));
var styleguide_1 = require("../styleguide");
var Typography_1 = __importDefault(require("../Typography/Typography"));
var utils_1 = require("../utils");
var Combobox_styles_1 = require("./Combobox.styles");
var ReactSelectComponents_1 = require("./ReactSelectComponents");
/** @description Combobox is built on top of `react-select`
 * @see [`react-select`](https://react-select.com/home) for more information
 */
var Combobox = function (_a) {
    var dataQA = _a.dataQA, placeholder = _a.placeholder, label = _a.label, labelHtmlFor = _a.labelHtmlFor, _b = _a.id, id = _b === void 0 ? (0, utils_1.uuid)() : _b, helperText = _a.helperText, options = _a.options, isMulti = _a.isMulti, comboboxIcon = _a.comboboxIcon, comboboxRef = _a.comboboxRef, required = _a.required, styles = _a.styles, validation = _a.validation, props = __rest(_a, ["dataQA", "placeholder", "label", "labelHtmlFor", "id", "helperText", "options", "isMulti", "comboboxIcon", "comboboxRef", "required", "styles", "validation"]);
    var customComponents = {
        Control: ReactSelectComponents_1.Control,
        Option: ReactSelectComponents_1.Option,
        MultiValue: ReactSelectComponents_1.MultiValue,
        ClearIndicator: ReactSelectComponents_1.ClearIndicator,
        GroupHeading: ReactSelectComponents_1.GroupHeading,
        DropdownIndicator: null,
        MultiValueRemove: function () { return react_1.default.createElement(react_1.default.Fragment, null); },
        LoadingMessage: function () { return react_1.default.createElement(Combobox_styles_1.LoadingIndicator, { dataQA: "".concat(dataQA, "-combobox-loading"), size: "Small" }); },
        NoOptionsMessage: function () { return (react_1.default.createElement(Combobox_styles_1.NoResultsFound, { dataQA: "comobobox-no-results-found", component: "p", variant: "bodyCopy", color: "".concat(styleguide_1.colors.black, "A8"), m: "0.625rem 0" }, "No results found.")); },
    };
    return (react_1.default.createElement(Combobox_styles_1.ComboboxWrapper, { m: props.m, mb: props.mb, mt: props.mt, mr: props.mr, ml: props.ml },
        label && (react_1.default.createElement(Label_1.default, { dataQA: "".concat(dataQA, "-select-label"), htmlFor: labelHtmlFor, m: "0 0 ".concat(styleguide_1.spacingMap.xxs, " 0"), isRequired: required, isInvalid: validation === "invalid" }, label)),
        react_1.default.createElement(react_select_1.default, __assign({ styles: (0, Combobox_styles_1.createStyles)(styles, validation), classNamePrefix: Combobox_styles_1.comboboxClassNamePrefix, placeholder: placeholder, inputId: id, options: options, hideSelectedOptions: false, closeMenuOnSelect: !isMulti, isMulti: isMulti, comboboxIcon: comboboxIcon, components: customComponents, ref: comboboxRef, required: required, validation: validation }, props)),
        helperText && (react_1.default.createElement(Typography_1.default, { dataQA: "".concat(dataQA, "-helper-text"), variant: "caption", component: "p", color: validation ? Combobox_styles_1.validationColor[validation] : "".concat(styleguide_1.colors.black, "A6"), m: "".concat(styleguide_1.spacingMap.xxs, "} 0 0 0") }, helperText))));
};
Combobox.displayName = "Combobox";
exports.default = Combobox;
