// @frontend/wknd-components/dist/src/Combobox/Combobox.styles.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionIcon = exports.OptionContentWrapper = exports.LoadingIndicator = exports.NoResultsFound = exports.ComboboxWrapper = exports.createStyles = exports.validationColor = exports.sharedCss = exports.comboboxClassNamePrefix = void 0;
var styled_components_1 = __importDefault(require("styled-components"));
var Icon_1 = require("../Icon/Icon");
var Spinner_1 = require("../Spinner");
var styleguide_1 = require("../styleguide");
var Typography_1 = __importDefault(require("../Typography/Typography"));
var utils_1 = require("../utils");
exports.comboboxClassNamePrefix = "Combobox";
exports.sharedCss = {
    fontFamily: styleguide_1.displayFontSizes.bodyCopy.$fontFamily,
    fontWeight: styleguide_1.displayFontSizes.bodyCopy.$fontWeight,
    fontSize: styleguide_1.displayFontSizes.bodyCopy.$fontSize,
    lineHeight: styleguide_1.displayFontSizes.bodyCopy.lineHeight,
};
exports.validationColor = {
    valid: styleguide_1.colors.green,
    invalid: styleguide_1.colors.destructive,
};
var createStyles = function (styles, validation) { return (__assign(__assign({}, styles), { control: function (provided, _a) {
        var isFocused = _a.isFocused, isDisabled = _a.isDisabled, selectProps = _a.selectProps, innerProps = _a.innerProps;
        return (__assign(__assign(__assign(__assign({}, provided), { minHeight: "40px", paddingLeft: "0.75rem", paddingRight: "0.75rem", borderWidth: "1px", borderStyle: "solid", borderRadius: "8px", color: validation ? exports.validationColor[validation] : styleguide_1.colors.defaultGray, borderColor: validation === "invalid" ? styleguide_1.colors.destructive : styleguide_1.colors.defaultGray, ":hover": {
                borderColor: styleguide_1.colors.black,
                color: validation ? exports.validationColor[validation] : styleguide_1.colors.black,
            } }), (isFocused && {
            boxShadow: "0 0 0 4px ".concat(styleguide_1.colors.midBlue, "40"),
            borderColor: validation === "invalid" ? styleguide_1.colors.destructive : styleguide_1.colors.midBlue,
            color: validation ? exports.validationColor[validation] : styleguide_1.colors.black,
            ":hover": {
                borderColor: validation === "invalid" ? styleguide_1.colors.destructive : styleguide_1.colors.midBlue,
            },
        })), (isDisabled && {
            background: styleguide_1.colors.lightestGray,
            borderColor: styleguide_1.colors.defaultGray,
            color: innerProps.defaultValue || !selectProps.placeholder
                ? "".concat(styleguide_1.colors.black, "A6")
                : "".concat(styleguide_1.colors.black, "4D"),
        })));
    }, placeholder: function (provided) { return (__assign(__assign(__assign({}, provided), exports.sharedCss), { color: styleguide_1.colors.defaultGray, margin: 0 })); }, input: function (provided) { return (__assign(__assign(__assign({}, provided), exports.sharedCss), { color: styleguide_1.colors.black, margin: 0, padding: 0, cursor: "text" })); }, valueContainer: function (provided, _a) {
        var _b = _a.selectProps, comboboxIcon = _b.comboboxIcon, comboboxLeftIcon = _b.comboboxLeftIcon;
        return (__assign(__assign(__assign({}, provided), exports.sharedCss), { padding: (comboboxIcon || comboboxLeftIcon) ? "".concat((0, utils_1.pxToRem)(10), " ").concat(styleguide_1.spacingMap.xs) : "".concat((0, utils_1.pxToRem)(10), " 0"), color: styleguide_1.colors.black, gap: styleguide_1.spacingMap.xs }));
    }, menu: function (provided, props) {
        var _a;
        return (__assign(__assign(__assign({}, provided), { marginTop: styleguide_1.spacingMap.xxs, boxShadow: "0 16px 32px -8px rgba(0, 0, 0, 0.08),"
                + "0 8px 8px rgba(0, 0, 0, 0.08),"
                + "0 4px 4px rgba(0, 0, 0, 0.08),"
                + "0 2px 2px rgba(0, 0, 0, 0.08)", borderRadius: "8px" }), ((_a = styles === null || styles === void 0 ? void 0 : styles.menu) === null || _a === void 0 ? void 0 : _a.call(styles, provided, props))));
    }, menuList: function (provided) { return (__assign(__assign({}, provided), { padding: styleguide_1.spacingMap.xs, gap: styleguide_1.spacingMap.xs, display: "flex", flexDirection: "column" })); }, option: function (provided, props) {
        var _a;
        return (__assign(__assign(__assign(__assign(__assign(__assign(__assign({}, provided), exports.sharedCss), { minHeight: "40px", display: "flex", alignItems: "center", justifyContent: "space-between", border: "2px solid transparent" }), (props.isFocused && {
            border: "2px solid ".concat(styleguide_1.colors.midBlue),
            borderRadius: "8px",
            cursor: "pointer",
        })), (props.isSelected && {
            color: styleguide_1.colors.midBlue,
            background: "none",
        })), { ":active": {
                background: "none",
            } }), ((_a = styles === null || styles === void 0 ? void 0 : styles.option) === null || _a === void 0 ? void 0 : _a.call(styles, provided, props))));
    }, multiValueLabel: function () { return ({}); }, multiValue: function () { return ({}); }, group: function (provided) { return (__assign(__assign({}, provided), { padding: 0 })); }, groupHeading: function (provided) { return (__assign(__assign({}, provided), { margin: 0 })); } })); };
exports.createStyles = createStyles;
exports.ComboboxWrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tdisplay: flex;\n\tflex-direction: column;\n\n\t", ";\n"], ["\n\tdisplay: flex;\n\tflex-direction: column;\n\n\t", ";\n"])), styleguide_1.marginsCss);
exports.NoResultsFound = (0, styled_components_1.default)(Typography_1.default)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\talign-self: center;\n"], ["\n\talign-self: center;\n"])));
exports.LoadingIndicator = (0, styled_components_1.default)(Spinner_1.Spinner)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\talign-self: center;\n"], ["\n\talign-self: center;\n"])));
exports.OptionContentWrapper = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tdisplay: flex;\n\talign-items: center;\n"], ["\n\tdisplay: flex;\n\talign-items: center;\n"])));
exports.OptionIcon = (0, styled_components_1.default)(Icon_1.Icon)(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\tmargin-right: ", ";\n"], ["\n\tmargin-right: ", ";\n"])), styleguide_1.spacingMap.xs);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
